_inc/client/components/navigation-settings/index.jsx

```diff
@@ -30,72 +28,60 @@ import { getSiteConnectionStatus } from 'state/connection';
 import { isModuleActivated } from 'state/modules';
 
 export const NavigationSettings = React.createClass( {
-	openSearch: function() {
-		let currentHash = window.location.hash;
-		if ( currentHash.indexOf( 'search' ) === -1 ) {
-			window.location.hash = 'search';
-		}
-		this.props.onSearchFocus && this.props.onSearchFocus( true );
-	},
-
-	onSearch( term ) {
-		if ( term.length >= 3 ) {
-			analytics.tracks.recordEvent( 'jetpack_wpa_search_term', { term: term.toLowerCase() } );
-		}
-
-		this.props.searchForTerm( trim( term || '' ).toLowerCase() );
+	mixins: [ UrlSearch ],
 
-		if ( 0 === term.length ) {
-
-			// Calling close handler to show what was previously shown to the user
-			this.onClose();
-		} else {
-
-			// Calling open handler in case the search was previously closed due to zero
-			// length search term
-			this.openSearch();
-		}
+	componentWillMount() {
+		this.context.router.listen( this.onRouteChange );
 	},
 
-	onClose: function() {
-		let currentHash = window.location.hash;
-		if ( currentHash.indexOf( 'search' ) > -1 ) {
-			this.context.router.goBack();
-		}
-	},
+	onRouteChange( newRoute ) {
+		const search = newRoute.search || '',
+			pairs = search.substr( 1 ).split( '&' ),
+			term = pairs.filter( item => {
+				return 0 === item.indexOf( 'term=' );
+			} );
 
-	onBlur: function() {
-		let currentHash = window.location.hash;
-		this.props.onSearchFocus( false );
+		let keyword = false;
 
-		// If the user has navigated back a page, we discard the search term
-		// on blur
-		if ( currentHash.indexOf( 'search' ) === -1 ) {
-			this.props.searchForTerm( false );
+		if ( term.length > 0 ) {
+			keyword = term[ 0 ].split( '=' )[ 1 ];
 		}
+
+		this.props.searchForTerm( keyword );
 	},
 
-	maybeShowSearch: function() {
+	maybeShowSearch() {
 		if ( this.props.userCanManageModules ) {
 			return (
 				<Search
 					pinned={ true }
+					fitsContainer={ true }
 					placeholder={ __( 'Search for a Jetpack feature.' ) }
 					delaySearch={ true }
 					delayTimeout={ 500 }
-					onSearchOpen={ this.openSearch }
-					onSearch={ this.onSearch }
-					onSearchClose={ this.onClose }
-					onBlur={ this.onBlur }
-					isOpen={
-						'/search' === this.props.route.path
-						|| this.props.searchHasFocus
-					}
+					onSearch={ this.doSearch }
+					isOpen={ false !== this.props.searchTerm }
+					initialValue={ this.props.searchTerm }
```

Comment:
I get this console warning when I load the page
```
Warning: Failed propType: Invalid prop `initialValue` of type `boolean` supplied to `Search`, expected `string`. Check the render method of `NavigationSettings`.
```
maybe this line?

I see that you're returning different types (boolean, string) in the reducer.